File: src/main/java/me/newyith/fortress/event/TickTimer.java
package me.newyith.fortress.event;

import me.newyith.fortress.bedrock.BedrockManager;
import me.newyith.fortress.bedrock.timed.TimedBedrockManager;
import me.newyith.fortress.command.Commands;
import me.newyith.fortress.main.FortressPlugin;
import me.newyith.fortress.main.FortressesManager;
import me.newyith.fortress.util.Debug;
import org.bukkit.scheduler.BukkitRunnable;

public class TickTimer extends BukkitRunnable {
	public static final int msPerTick = 150; //should be divisible by 50
	//tick internally only every 3rd game tick in hopes it will improve performance

	public static void onEnable(FortressPlugin plugin) {
		new TickTimer().runTaskTimer(plugin, 0, msPerTick / 50);
	}

	@Override
	public void run() {
		FortressesManager.onTick();
		TimedBedrockManager.onTick();
		BedrockManager.onTick();
		Commands.onTick();
		FortressPlugin.onTick();
	}
}

File: src/main/java/me/newyith/fortress/event/TickTimer.kt
package me.newyith.fortress.event

import me.newyith.fortress.main.FortressPlugin
import org.bukkit.plugin.java.JavaPlugin
import org.bukkit.scheduler.BukkitRunnable

class TickTimer(plugin: JavaPlugin) : BukkitRunnable() {
	init {
		this.runTaskTimer(plugin, 0, (msPerTick / 50).toLong())
	}

	override fun run() {
		//Log.start("globalsTick");
//		Commands.onTick() //TODO: uncomment out this line
		FortressPlugin.onTick()
		//Log.end("globalsTick");
	}

	companion object {
		val msPerTick = 150 //should be divisible by 50
		//tick internally only every 3rd game tick in hopes it will improve performance
	}
}